### DISCUSSION TITLE
Debouncing/caching strategies in serverless functions

### DISCUSSION THREAD

**Discussion by stevefrench39** [vercel/vercel#5125] [Help]
Hi,

I have an endpoint that sends an email. It should only ever send a single email message despite potentially being called many times over a very short timespan.

What are some ways to accomplish this in a serverless function?

Thanks in advance for your advice!

**Comment by styfle** [comment]
Hi @networkchimp 

It sounds like you should use POST or PUT to avoid multiple calls, which is not cached by HTTP Clients, unlike GET which can be cached.

If you still need to debounce, you can use an external datastore like Redis for storing and sharing state between your Serverless Functions.
